pkg/trace/config/config.go

```diff
@@ -143,8 +143,8 @@ func New() *AgentConfig {
 
 		MaxMemory:        5e8, // 500 Mb, should rarely go above 50 Mb
 		MaxCPU:           0.5, // 50%, well behaving agents keep below 5%
-		MaxConnections:   200, // in practice, rarely goes over 20
-		WatchdogInterval: time.Minute,
+		MaxConnections:   200, // (deprecated) - use ConnectionLimit
+		WatchdogInterval: 20 * time.Second,
```

Comment:
I agree the interval was likely too short. But what was the rational behind `20s`? why not `10s`? `5s`?

Reply:
No rationale. It was just a value that I've decided to test with and considered that any shorter than this could potentially be felt since `ReadMemStats` causes "stop the world" behavior in the runtime.

Reply:
How bad are the pauses? If they're not too bad could be useful to do it more frequently, else it could end up with more idle time than necessary.

Reply:
We've cut the time in 3, and we've proven this to work well in production. I would go as high as possible without taking too long between checks. One thing you might've missed is the way the presampler works and that it only stops a percentage of payloads based on how much we've gone over the CPU threshold, it's not a "reject every payload" state. The subsequent PR I'm talking about in other comments will make the memory presampler work exactly the same.